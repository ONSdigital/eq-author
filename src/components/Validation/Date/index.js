import dateBuilder from "./builder";
import LatestDateValidationRule from "graphql/fragments/latest-date-validation-rule.graphql";
import EarliestDateValidationRule from "graphql/fragments/earliest-date-validation-rule.graphql";

export const LatestDate = dateBuilder(
  "Latest date",
  "latest-date-validation",
  "latestDate",
  "latestDateInput",
  LatestDateValidationRule
);

export const EarliestDate = dateBuilder(
  "Earliest date",
  "earliest-date-validation",
  "earliestDate",
  "earliestDateInput",
  EarliestDateValidationRule
);
