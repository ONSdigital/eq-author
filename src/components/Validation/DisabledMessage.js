import styled from "styled-components";

const DisabledMessage = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex: 1 1 auto;
  font-weight: bold;
  margin-bottom: 3em;
`;

export default DisabledMessage;
