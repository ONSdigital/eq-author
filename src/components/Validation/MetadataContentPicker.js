import React from "react";
import PropTypes from "prop-types";
import { get } from "lodash";

import ContentPickerSelect from "components/ContentPickerSelect/index";
import { METADATA } from "components/ContentPickerSelect/content-types";

import AvailableMetadataQuery from "components/Validation/AvailableMetadataQuery";

export const UnwrappedMetadataContentPicker = ({
  data,
  path,
  ...otherProps
}) => (
  <ContentPickerSelect
    name="metadata"
    contentTypes={[METADATA]}
    metadataData={get(data, path)}
    {...otherProps}
  />
);

UnwrappedMetadataContentPicker.propTypes = {
  data: PropTypes.object, // eslint-disable-line
  path: PropTypes.string.isRequired
};

export default props => (
  <AvailableMetadataQuery answerId={props.answerId}>
    {innerProps => (
      <UnwrappedMetadataContentPicker {...innerProps} {...props} />
    )}
  </AvailableMetadataQuery>
);
