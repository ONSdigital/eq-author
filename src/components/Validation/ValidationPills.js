import CustomPropTypes from "custom-prop-types";
import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { get } from "lodash";

import PillTabs from "components/PillTabs";
import ContentPickerSelect from "components/ContentPickerSelect";

import ValidationInput from "components/Validation/ValidationInput";

import * as entityTypes from "constants/validation-entity-types";
import * as answerTypes from "constants/answer-types";

const Pills = styled(PillTabs)`
  width: 5em;
`;

const PillTabContent = styled.div`
  margin-top: 3em;
  margin-bottom: 0;
  width: 100%;
`;

export const ValidationPills = ({
  entityType,
  answerType,
  previousAnswer,
  customValue,
  customValueLimit,
  onEntityTypeChange,
  onPreviousAnswerChange,
  onCustomValueChange
}) => (
  <Pills
    value={entityType}
    onChange={onEntityTypeChange}
    data-test="max-value-pill-tabs"
    options={[
      {
        id: "PreviousAnswer",
        title: "Previous answer",
        render: () => (
          <PillTabContent>
            <ContentPickerSelect
              onSubmit={onPreviousAnswerChange}
              answerTypes={[answerType]}
              selectedContentDisplayName={get(previousAnswer, "displayName")}
            />
          </PillTabContent>
        )
      },
      {
        id: "Custom",
        title: "Custom",
        render: () => (
          <PillTabContent>
            <ValidationInput
              data-test="max-value-input"
              list="defaultNumbers"
              value={customValue}
              type="number"
              id="max-value"
              onChange={onCustomValueChange}
              max={customValueLimit}
              min={0 - customValueLimit}
            />
          </PillTabContent>
        )
      }
    ]}
  />
);

ValidationPills.propTypes = {
  entityType: PropTypes.oneOf(Object.values(entityTypes)).isRequired,
  answerType: PropTypes.oneOf(Object.values(answerTypes)).isRequired,
  previousAnswer: PropTypes.shape({
    displayName: CustomPropTypes.string
  }),
  customValue: PropTypes.number,
  customValueLimit: PropTypes.number.isRequired,
  onEntityTypeChange: PropTypes.func.isRequired,
  onPreviousAnswerChange: PropTypes.func.isRequired,
  onCustomValueChange: PropTypes.func.isRequired
};
