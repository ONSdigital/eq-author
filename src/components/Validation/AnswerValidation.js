import React from "react";
import { kebabCase, get } from "lodash";
import styled from "styled-components";
import CustomPropTypes from "custom-prop-types";
import PropTypes from "prop-types";
import { connect } from "react-redux";

import { gotoTab } from "redux/tabs/actions";
import { CURRENCY, DATE, NUMBER } from "constants/answer-types";
import { colors } from "constants/theme";

import SidebarButton, { Title, Detail } from "components/SidebarButton";
import ModalWithNav from "components/ModalWithNav";
import MinValueValidation from "./MinValue";
import MaxValueValidation from "./MaxValue";
import { EarliestDate, LatestDate } from "./Date";

import ValidationContext from "./ValidationContext";

const Container = styled.div`
  margin-top: 1em;
  border-top: 1px solid ${colors.lightGrey};
  padding: 1em 0;
`;

const formatDate = dateString =>
  dateString
    .split("-")
    .reverse()
    .join("/");

const DatePreview = ({
  relativePosition,
  customDate,
  offset: { unit, value }
}) => {
  if (!customDate) {
    return "Invalid Rule";
  }

  const formattedDate = formatDate(customDate);
  if (value === 0) {
    return formattedDate;
  }
  return (
    <React.Fragment>
      <div>
        {`${value} ${unit.toLowerCase()} ${relativePosition.toLowerCase()}:`}
      </div>
      <div>{formattedDate}</div>
    </React.Fragment>
  );
};
DatePreview.propTypes = {
  relativePosition: PropTypes.string.isRequired,
  customDate: PropTypes.string,
  offset: PropTypes.shape({
    value: PropTypes.number.isRequired,
    unit: PropTypes.string.isRequired
  }).isRequired
};

const validationTypes = [
  {
    id: "minValue",
    title: "Min Value",
    render: () => <MinValueValidation />,
    types: [CURRENCY, NUMBER],
    preview: ({ custom }) => custom
  },
  {
    id: "maxValue",
    title: "Max Value",
    render: () => <MaxValueValidation />,
    types: [CURRENCY, NUMBER],
    preview: ({ custom }) => custom
  },
  {
    id: "earliestDate",
    title: "Earliest Date",
    render: () => <EarliestDate />,
    types: [DATE],
    preview: DatePreview
  },
  {
    id: "latestDate",
    title: "Latest Date",
    render: () => <LatestDate />,
    types: [DATE],
    preview: DatePreview
  }
];

const getValidationsForType = type =>
  validationTypes.filter(({ types }) => types.includes(type));

const validations = [NUMBER, CURRENCY, DATE].reduce(
  (hash, type) => ({
    ...hash,
    [type]: getValidationsForType(type)
  }),
  {}
);

export class UnconnectedAnswerValidation extends React.Component {
  state = {
    modalIsOpen: false,
    answerId: null
  };

  constructor(props) {
    super(props);
    this.modalId = `modal-validation-${props.answer.id}`;
  }

  handleModalClose = () => this.setState({ modalIsOpen: false });

  renderButton = ({ id, title, value, enabled }) => (
    <SidebarButton
      key={id}
      data-test={`sidebar-button-${kebabCase(title)}`}
      onClick={() => {
        this.props.gotoTab(this.modalId, id);
        this.setState({ modalIsOpen: true });
      }}
    >
      <Title>{title}</Title>
      {enabled && value && <Detail>{value}</Detail>}
    </SidebarButton>
  );

  render() {
    const { answer } = this.props;

    const validValidationTypes = validations[answer.type] || [];

    if (validValidationTypes.length === 0) {
      return null;
    }

    return (
      <Container>
        <ValidationContext.Provider value={{ answer }}>
          {validValidationTypes.map(validationType => {
            const validation = get(
              answer,
              `validation.${validationType.id}`,
              {}
            );
            const { enabled } = validation;
            const value = enabled ? validationType.preview(validation) : "";
            return this.renderButton({
              ...validationType,
              value,
              enabled
            });
          })}
          <ModalWithNav
            id={this.modalId}
            onClose={this.handleModalClose}
            navItems={validValidationTypes}
            title={`${answer.type} validation`}
            isOpen={this.state.modalIsOpen}
          />
        </ValidationContext.Provider>
      </Container>
    );
  }
}

UnconnectedAnswerValidation.propTypes = {
  answer: CustomPropTypes.answer,
  gotoTab: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  tabsState: state.tabs
});

export default connect(
  mapStateToProps,
  { gotoTab }
)(UnconnectedAnswerValidation);
