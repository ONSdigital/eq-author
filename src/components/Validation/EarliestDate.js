import { propType } from "graphql-anywhere";
import { flowRight } from "lodash/fp";
import PropTypes from "prop-types";
import React from "react";
import styled from "styled-components";

import EarliestDateValidationRule from "graphql/fragments/earliest-date-validation-rule.graphql";

import { Input, Number, Select } from "components/Forms";
import { Grid, Column } from "components/Grid";
import DisabledMessage from "components/Validation/DisabledMessage";

import withToggleAnswerValidation from "containers/enhancers/withToggleAnswerValidation";
import withUpdateAnswerValidation from "containers/enhancers/withUpdateAnswerValidation";
import withEntityEditor from "components/withEntityEditor";

import ValidationView from "./ValidationView";
import withAnswerValidation from "./withAnswerValidation";

import svgPath from "./path.svg";
import svgPathEnd from "./path-end.svg";

const UNITS = ["Days", "Months", "Years"];
const RELATIVE_POSITIONS = ["Before", "After"];

const DateInput = styled(Input)`
  margin-top: 4em;
  width: 12em;
`;
const ConnectedPath = styled.div`
  padding-left: 1em;
  position: relative;
  height: 100%;

  &::after {
    position: absolute;
    content: "";
    background: url(${({ pathUrl }) => pathUrl}) no-repeat center center;
    background-size: auto;
    width: 100%;
    height: calc(100% - 2em);
    top: 0;
    bottom: 0;
    margin: auto;
  }
`;

const ConnectedPathStart = styled(ConnectedPath)`
  height: 5em;
`;

const AlignedColumn = styled(Column)`
  align-items: center;
  display: flex;
  justify-content: center;
`;

const RelativePositionSelect = styled(Select)`
  width: 6em;
`;

const EmphasisedText = styled.p`
  font-size: 0.9em;
  font-weight: bold;
  text-transform: uppercase;
`;

const START_COL_SIZE = 3;
const END_COL_SIZE = 12 - START_COL_SIZE;

export const EarliestDate = ({
  earliestDate,
  onToggleValidationRule,
  onUpdate,
  onChange
}) => {
  const { id, enabled, customDate, offset, relativePosition } = earliestDate;

  const handleToggleChange = ({ value: enabled }) => {
    onToggleValidationRule({
      id,
      enabled
    });
  };
  const renderContent = () => (
    <div>
      <Grid>
        <AlignedColumn cols={START_COL_SIZE}>
          <EmphasisedText>Earliest date is</EmphasisedText>
        </AlignedColumn>
        <Column cols={END_COL_SIZE}>
          <Grid>
            <Column cols={2}>
              <Number
                id="offset-value"
                name="offset.value"
                value={offset.value}
                onChange={onChange}
                onBlur={onUpdate}
                max={99999}
                min={0}
              />
            </Column>
            <Column cols={3}>
              <Select
                name="offset.unit"
                value={offset.unit}
                onChange={onChange}
                onBlur={onUpdate}
                data-test="offset-unit-select"
              >
                {UNITS.map(unit => (
                  <option key={unit} value={unit}>
                    {unit}
                  </option>
                ))}
              </Select>
            </Column>
          </Grid>
        </Column>
      </Grid>
      <Grid>
        <Column cols={START_COL_SIZE}>
          <ConnectedPathStart pathUrl={svgPath} />
        </Column>
      </Grid>
      <Grid>
        <AlignedColumn cols={START_COL_SIZE}>
          <RelativePositionSelect
            name="relativePosition"
            value={relativePosition}
            onChange={onChange}
            onBlur={onUpdate}
            data-test="relative-position-select"
          >
            {RELATIVE_POSITIONS.map(position => (
              <option key={position} value={position}>
                {position}
              </option>
            ))}
          </RelativePositionSelect>
        </AlignedColumn>
      </Grid>
      <Grid>
        <Column cols={START_COL_SIZE}>
          <ConnectedPath pathUrl={svgPathEnd} />
        </Column>
        <Column cols={END_COL_SIZE}>
          <DateInput
            name="customDate"
            type="date"
            value={customDate}
            onChange={onChange}
            onBlur={onUpdate}
          />
        </Column>
      </Grid>
    </div>
  );

  const renderDisabled = () => (
    <DisabledMessage>Earliest date is disabled</DisabledMessage>
  );

  return (
    <ValidationView
      data-test="earliest-date-validation"
      enabled={enabled}
      onToggleChange={handleToggleChange}
    >
      {enabled ? renderContent() : renderDisabled()}
    </ValidationView>
  );
};

EarliestDate.propTypes = {
  earliestDate: propType(EarliestDateValidationRule).isRequired,
  onToggleValidationRule: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  onUpdate: PropTypes.func.isRequired
};

export const readToWriteMapper = ({ id, customDate, enabled, ...rest }) => ({
  id,
  earliestDateInput: {
    ...rest,
    custom: customDate ? customDate : null
  }
});

export const remapToOnUpdate = (propName, mapper) => Component => props => (
  <Component
    onUpdate={(...args) => props[propName](mapper(...args))}
    {...props}
  />
);

const withEditing = flowRight(
  withAnswerValidation("earliestDate"),
  withUpdateAnswerValidation,
  withToggleAnswerValidation,
  remapToOnUpdate("onUpdateAnswerValidation", readToWriteMapper),
  withEntityEditor("earliestDate", EarliestDateValidationRule)
);
export default withEditing(EarliestDate);
