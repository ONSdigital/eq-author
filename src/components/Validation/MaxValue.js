import React from "react";
import PropTypes from "prop-types";

import styled from "styled-components";
import { withApollo } from "react-apollo";
import { propType } from "graphql-anywhere";

import { flowRight, inRange } from "lodash";

import { colors } from "constants/theme";
import { Grid, Column } from "components/Grid";
import { Input } from "components/Forms";
import DisabledMessage from "components/Validation/DisabledMessage";

import ValidationView from "./ValidationView";
import FieldWithInclude from "./FieldWithInclude";

import MaxValueValidationRule from "graphql/fragments/max-value-validation-rule.graphql";
import withUpdateAnswerValidation from "containers/enhancers/withUpdateAnswerValidation";
import withToggleAnswerValidation from "containers/enhancers/withToggleAnswerValidation";

import ValidationContext from "./ValidationContext";

const Title = styled.h1`
  font-weight: bold;
  letter-spacing: 0.05em;
  text-transform: uppercase;
  font-size: 0.9em;
  text-align: center;
  color: ${colors.text};
  margin: 0.6em 0 0;
`;

const ValueInput = styled(Input)`
  width: 10em;
`;

export const MaxValue = ({
  maxValue,
  onUpdateAnswerValidation,
  onToggleValidationRule,
  limit
}) => {
  const handleToggleChange = ({ value }) => {
    const toggleValidationRuleInput = {
      id: maxValue.id,
      enabled: value
    };

    onToggleValidationRule(toggleValidationRuleInput);
  };

  const handleMaxValueChange = ({ value }) => {
    // clamp value of input to +/- limit
    if (value !== "" && !inRange(parseInt(value, 10), 0 - limit, limit + 1)) {
      return false;
    }

    const updateValidationRuleInput = {
      id: maxValue.id,
      maxValueInput: {
        inclusive: maxValue.inclusive,
        custom: value === "" ? null : value
      }
    };

    onUpdateAnswerValidation(updateValidationRuleInput);
  };

  const handleIncludeChange = ({ value }) => {
    const updateValidationRuleInput = {
      id: maxValue.id,
      maxValueInput: {
        custom: maxValue.custom,
        inclusive: value
      }
    };
    onUpdateAnswerValidation(updateValidationRuleInput);
  };

  const renderDisabled = () => (
    <DisabledMessage>Max value is disabled</DisabledMessage>
  );

  const renderContent = () => (
    <Grid>
      <Column cols={3}>
        <Title>Max Value is</Title>
      </Column>
      <Column>
        <FieldWithInclude
          id="max-value-include"
          name="max-value-include"
          onChange={handleIncludeChange}
          checked={maxValue.inclusive}
        >
          <ValueInput
            data-test="max-value-input"
            list="defaultNumbers"
            value={maxValue.custom}
            type="number"
            id="max-value"
            onChange={handleMaxValueChange}
            max={limit}
            min={0 - limit}
          />
        </FieldWithInclude>
      </Column>
    </Grid>
  );

  return (
    <ValidationView
      onToggleChange={handleToggleChange}
      enabled={maxValue.enabled}
      data-test="max-value-view"
    >
      {maxValue.enabled ? renderContent() : renderDisabled()}
    </ValidationView>
  );
};

MaxValue.defaultProps = {
  limit: 999999999
};

MaxValue.propTypes = {
  maxValue: propType(MaxValueValidationRule).isRequired,
  onUpdateAnswerValidation: PropTypes.func.isRequired,
  onToggleValidationRule: PropTypes.func.isRequired,
  limit: PropTypes.number
};

const withQuestionPageEditing = flowRight(
  withApollo,
  withUpdateAnswerValidation,
  withToggleAnswerValidation
);

export const MaxValueWithAnswer = props => (
  <ValidationContext.Consumer>
    {({ answer }) => (
      <MaxValue maxValue={answer.validation.maxValue} {...props} />
    )}
  </ValidationContext.Consumer>
);

export default withQuestionPageEditing(MaxValueWithAnswer);
