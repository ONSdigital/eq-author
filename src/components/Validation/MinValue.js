import React from "react";
import PropTypes from "prop-types";

import styled from "styled-components";
import { withApollo } from "react-apollo";
import { propType } from "graphql-anywhere";

import { flowRight, inRange, isNaN } from "lodash";

import { colors } from "constants/theme";
import { Grid, Column } from "components/Grid";
import { Input } from "components/Forms";
import DisabledMessage from "components/Validation/DisabledMessage";

import ValidationView from "./ValidationView";
import FieldWithInclude from "./FieldWithInclude";

import MinValueValidationRule from "graphql/fragments/min-value-validation-rule.graphql";
import withUpdateAnswerValidation from "containers/enhancers/withUpdateAnswerValidation";
import withToggleAnswerValidation from "containers/enhancers/withToggleAnswerValidation";

import ValidationContext from "./ValidationContext";

const Title = styled.h1`
  font-weight: bold;
  letter-spacing: 0.05em;
  text-transform: uppercase;
  font-size: 0.9em;
  text-align: center;
  color: ${colors.text};
  margin: 0.6em 0 0;
`;

const ValueInput = styled(Input)`
  width: 10em;
`;

export const MinValue = ({
  minValue,
  onUpdateAnswerValidation,
  onToggleValidationRule,
  limit
}) => {
  const handleToggleChange = ({ value }) => {
    const toggleValidationRuleInput = {
      id: minValue.id,
      enabled: value
    };

    onToggleValidationRule(toggleValidationRuleInput);
  };

  const handleMinValueChange = ({ value }) => {
    // clamp value of input to +/- limit
    if (value !== "" && !inRange(parseInt(value, 10), 0 - limit, limit + 1)) {
      return false;
    }

    const intValue = parseInt(value, 10);

    const updateValidationRuleInput = {
      id: minValue.id,
      minValueInput: {
        inclusive: minValue.inclusive,
        custom: isNaN(intValue) ? null : intValue
      }
    };

    onUpdateAnswerValidation(updateValidationRuleInput);
  };

  const handleIncludeChange = ({ value }) => {
    const updateValidationRuleInput = {
      id: minValue.id,
      minValueInput: {
        custom: minValue.custom,
        inclusive: value
      }
    };
    onUpdateAnswerValidation(updateValidationRuleInput);
  };

  const renderDisabled = () => (
    <DisabledMessage>Min value is disabled</DisabledMessage>
  );

  const renderContent = () => (
    <Grid>
      <Column cols={3}>
        <Title>Min Value is</Title>
      </Column>
      <Column>
        <FieldWithInclude
          id="min-value-include"
          name="min-value-include"
          onChange={handleIncludeChange}
          checked={minValue.inclusive}
        >
          <ValueInput
            data-test="min-value-input"
            list="defaultNumbers"
            value={minValue.custom}
            type="number"
            id="min-value"
            onChange={handleMinValueChange}
            max={limit}
            min={0 - limit}
          />
        </FieldWithInclude>
      </Column>
    </Grid>
  );

  return (
    <ValidationView
      onToggleChange={handleToggleChange}
      enabled={minValue.enabled}
      data-test="min-value-view"
    >
      {minValue.enabled ? renderContent() : renderDisabled()}
    </ValidationView>
  );
};

MinValue.defaultProps = {
  limit: 999999999
};

MinValue.propTypes = {
  minValue: propType(MinValueValidationRule).isRequired,
  onUpdateAnswerValidation: PropTypes.func.isRequired,
  onToggleValidationRule: PropTypes.func.isRequired,
  limit: PropTypes.number
};

const withQuestionPageEditing = flowRight(
  withApollo,
  withUpdateAnswerValidation,
  withToggleAnswerValidation
);

export const MinValueWithAnswer = props => (
  <ValidationContext.Consumer>
    {({ answer }) => (
      <MinValue minValue={answer.validation.minValue} {...props} />
    )}
  </ValidationContext.Consumer>
);

export default withQuestionPageEditing(MinValueWithAnswer);
