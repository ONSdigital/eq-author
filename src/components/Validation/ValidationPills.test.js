import React from "react";
import { shallow } from "enzyme";

import { Pills, ValidationPills } from "components/Validation/ValidationPills";

import { PREVIOUS_ANSWER } from "constants/validation-entity-types";

const createWrapper = (props, render = shallow) =>
  render(<ValidationPills {...props} />);

describe("ValidationPills", () => {
  let props, wrapper;

  beforeEach(() => {
    props = {
      entityType: PREVIOUS_ANSWER,
      onEntityTypeChange: jest.fn(),
      PreviousAnswer: () => <div>Previous Answer Content</div>,
      Custom: () => <div>Previous Answer Content</div>
    };

    wrapper = createWrapper(props);
  });

  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render previous answer", () => {
    const previousAnswer = wrapper.find(Pills).prop("options")[0];
    expect(shallow(previousAnswer.render())).toMatchSnapshot();
  });

  it("should render custom", () => {
    const custom = wrapper.find(Pills).prop("options")[1];
    expect(shallow(custom.render())).toMatchSnapshot();
  });
});
