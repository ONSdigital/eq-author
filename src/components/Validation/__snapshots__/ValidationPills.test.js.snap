// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidationPills should render 1`] = `
<ValidationPills__Pills
  data-test="max-value-pill-tabs"
  onChange={[MockFunction]}
  options={
    Array [
      Object {
        "id": "PreviousAnswer",
        "render": [Function],
        "title": "Previous answer",
      },
      Object {
        "id": "Custom",
        "render": [Function],
        "title": "Custom",
      },
    ]
  }
  value="PreviousAnswer"
/>
`;